src/main/scala/com/thirdship/libunit/AStarSolver.scala

```diff
@@ -101,29 +103,29 @@ case class AStarSolver(var allTSUnits: List[String], var allConversions: List[Co
       * @param  end      The ending unit form the search
       * @return A list of conversion edges that when listed in order describe the path of conversion from start to end
 	  */
-  private def reconstructConversionEdgePathList(cameFrom: Map[ String, String], end: String): List[ConversionEdge[String, Double, Double]] = {
+  private def reconstructConversionEdgePathList(cameFrom: mutable.Map[String, String], end: String): List[ConversionEdge[String, Double, Double]] = {
```

Comment:
As addressed last night, I will change it all back to immutable maps. Also, I will update the ScalaStyle sheet to bring the maximum number of lines allowed in a function to a higher number.